esmvaltool/_recipe.py

```diff
@@ -758,6 +774,10 @@ def _initialize_variables(self, raw_variable, raw_datasets):
             check_variable(variable, required_keys)
             variable['filename'] = get_output_file(variable,
                                                    self._cfg['preproc_dir'])
+            if 'fx_variable' in variable.keys():
+                # Get the fx files
+                variable['fx_files'] = _get_input_fx_files(variable,
```

Comment:
The function `_get_input_fx_files` is not needed, you can directly call `get_input_fx_filelist` from datafinder, now everything is added double, e.g.:
```
fx_files:
  orog:
    orog: [/home/bandela/esmvaltool_input/orog/orog_fx_bcc-csm1-1_historical_r0i0p0.nc]
    sftlf: [/home/bandela/esmvaltool_input/sftlf/sftlf_fx_bcc-csm1-1_historical_r0i0p0.nc]
  sftlf:
    orog: [/home/bandela/esmvaltool_input/orog/orog_fx_bcc-csm1-1_historical_r0i0p0.nc]
    sftlf: [/home/bandela/esmvaltool_input/sftlf/sftlf_fx_bcc-csm1-1_historical_r0i0p0.nc]
```